lib/src/tes/mod.rs

```diff
@@ -0,0 +1,328 @@
+            .await;
+        match response {
+            Ok(response_body) => {
+                let v: serde_json::Value = serde_json::from_str(&response_body)?;
+
+                // Access the `id` field
+                let task_id = v
+                    .get("id")
+                    .and_then(|v| v.as_str())
+                    .unwrap_or_default()
+                    .trim_matches('"')
+                    .to_string();
+
+                let task = Task {
+                    id: task_id,
+                    transport: self.transport.clone(),
+                };
+                Ok(task)
+            }
+            Err(e) => Err(Box::new(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                format!("Failed to post task: {}", e),
+            ))),
+        }
+    }
+
+    pub async fn get(&self, view: &str, id: &str) -> Result<TesTask, Box<dyn std::error::Error>> {
+        let task_id = id;
+        let url = format!("/tasks/{}?view={}", task_id, view);
+        // let params = [("view", view)];
+        // let params_value = serde_json::json!(params);
+        // let response = self.transport.get(&url, Some(params_value)).await;
+        let response = self.transport.get(&url, None).await;
+        match response {
+            Ok(resp_str) => {
+                let task: TesTask = from_str(&resp_str)?;
+                Ok(task)
+            }
+            Err(e) => Err(e),
+        }
+    }
+    pub async fn list_tasks(
+        &self,
+        params: Option<ListTasksParams>,
+    ) -> Result<TesListTasksResponse, Box<dyn std::error::Error>> {
+        let params_value = params.map(|p| {
+            let mut map = serde_json::Map::new();
+            
+            if let Some(name_prefix) = p.name_prefix {
+                map.insert("name_prefix".to_string(), json!(name_prefix));
+            }
+            if let Some(state) = p.state {
+                map.insert("state".to_string(), json!(state));
+            }
+            if let Some(tag_key) = p.tag_key {
+                map.insert("tag_key".to_string(), json!(tag_key));
+            }
+            if let Some(tag_value) = p.tag_value {
+                map.insert("tag_value".to_string(), json!(tag_value));
+            }
+            if let Some(page_size) = p.page_size {
+                map.insert("page_size".to_string(), json!(page_size));
+            }
+            if let Some(page_token) = p.page_token {
+                map.insert("page_token".to_string(), json!(page_token));
+            }
+            if let Some(view) = p.view {
+                map.insert("view".to_string(), json!(view));
+            }
+
+            json!(map)
+        });
+        // println!("{:?}",params_value);
+        // Make the request with or without parameters based on the presence of params
+        let response = if let Some(params_value) = params_value {
+            self.transport.get("/tasks", Some(params_value)).await
```

Comment:
I believe we have to find a way to implement Transport methods (e.g., `get`) in a way that they can accept any struct as a URL parameter for the provided end-point, like this:

```
    pub async fn list_tasks(
        &self,
        params: Option<ListTasksParams>,
    ) -> Result<TesListTasksResponse, Box<dyn std::error::Error>> {
           self.transport.get("/tasks", params).await;
```
           
The postprocessing of `params` should be done inside Transport, including removing the Null parameters as they cause errors. 

Otherwise, there will be a need to do such complex and custom workarounds whenever we pass parameters to the Trasnport methods